pkg/controllers/node/health/controller.go

```diff
@@ -155,11 +164,25 @@ func (c *Controller) findUnhealthyConditions(node *corev1.Node) (nc *corev1.Node
 }
 
 func (c *Controller) annotateTerminationGracePeriod(ctx context.Context, nodeClaim *v1.NodeClaim) error {
+	if expirationTimeString, exists := nodeClaim.ObjectMeta.Annotations[v1.NodeClaimTerminationTimestampAnnotationKey]; exists {
+		expirationTime, err := time.Parse(time.RFC3339, expirationTimeString)
+		if err != nil {
+			return err
+		}
+		if c.clock.Now().Before(expirationTime) {
```

Comment:
shouldn't this be the opposite? 

if the current time is before the expiration time, we actually do want to annotate it. If it isn't, then we keep it as is. 

Can you add a test for this?

Reply:
Yeah, this was the wrong way around, and added a test to validate the behavior